app/code/Magento/InventoryInStorePickup/Model/PickupLocation/Mapper/CreateFromSource.php

```diff
@@ -0,0 +1,123 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\InventoryInStorePickup\Model\PickupLocation\Mapper;
+
+use Magento\Framework\Api\ExtensionAttributesFactory;
+use Magento\Framework\Api\SimpleDataObjectConverter;
+use Magento\InventoryApi\Api\Data\SourceInterface;
+use Magento\InventoryInStorePickup\Model\PickupLocationFactory;
+use Magento\InventoryInStorePickupApi\Api\Data\PickupLocationInterface;
+
+class CreateFromSource
+{
+    /**
+     * @var \Magento\InventoryInStorePickup\Model\PickupLocationFactory
+     */
+    private $pickupLocationFactory;
+
+    /**
+     * @var \Magento\Framework\Api\ExtensionAttributesFactory
+     */
+    private $extensionAttributesFactory;
+
+    /**
+     * CreateFromSource constructor.
+     *
+     * @param \Magento\InventoryInStorePickup\Model\PickupLocationFactory $pickupLocationFactory
+     * @param \Magento\Framework\Api\ExtensionAttributesFactory $extensionAttributesFactory
+     */
+    public function __construct(
+        PickupLocationFactory $pickupLocationFactory,
+        ExtensionAttributesFactory $extensionAttributesFactory
+    ) {
+        $this->pickupLocationFactory = $pickupLocationFactory;
+        $this->extensionAttributesFactory = $extensionAttributesFactory;
+    }
+
+    /**
+     * @param \Magento\InventoryApi\Api\Data\SourceInterface $source
+     * @param array $map
+     *
+     * @return \Magento\InventoryInStorePickup\Model\PickupLocation
+     */
+    public function execute(SourceInterface $source, array $map)
+    {
+        $data = [];
+        $pickupLocationExtension = $this->extensionAttributesFactory->create(PickupLocationInterface::class);
+
+        foreach ($map as $sourceField => $pickupLocationField) {
+            if ($this->isExtensionAttributeField($sourceField)) {
+                $fieldValue = $source->getExtensionAttributes()->{$this->getGetterMethodName(
```

Comment:
Method may be not found in case someone misconfigures map

Reply:
Added exception throwing in such case.